app-cpp/app355.cpp

```diff
@@ -393,6 +324,81 @@ void SatOrbit::OnSortOrbitalVectorMulti(orbit_iterator& inBegin, orbit_iterator&
     return inLHS.GetTLE().GetMeanMotion() < inRHS.GetTLE().GetMeanMotion();
 }
 
+void SatOrbit::SortOrbitalVectorMulti(std::vector<OrbitalData>& ioOrbitalVector)
+{
+    const std::size_t orbitVectorSize = ioOrbitalVector.size();
+
+        // calculate the number of TLEs per thread
+        const std::size_t orbitPerThread = orbitVectorSize / mNumThreads;
+        // Note: no need for mutex as threads do not contend for shared elements during sort
+        IteratorPairVector subListVector{};
+
+        {
+            // create a list of threads the size of the number of threads specified
+            std::vector<std::thread> threadVector{};
+
+            for (std::size_t i = 0; i < mNumThreads; ++i)
+            {
+                // calculate the begin and end of the TLEs for the current thread
+                const auto orbitBegin = ioOrbitalVector.begin() + (i * orbitPerThread);
+                // if this is the last thread, add the remaining TLEs, else add orbitPerThread
+                const auto orbitEnd = (i == mNumThreads - 1) ? ioOrbitalVector.end() : orbitBegin + orbitPerThread;
+                threadVector.push_back(std::thread(&SatOrbit::OnSortOrbitalVectorMulti, this, orbitBegin, orbitEnd));
+
+                IteratorPairVector::value_type slice{ orbitBegin, orbitEnd };
```

Comment:
Use `std::make_tuple()` here, instead of default ctor{}.
That way it'll become:
```
    auto slice = std::make_tuple(...
```
Notice how it is `auto slice =` and not `const auto slice =`...
The reason is to allow comment on next statement!